### PR TITLE
feat: added useSearch composables

### PR DESCRIPTION
**Pull request by sevensilvers**
This is a feature where it could search through products from Shopify API by the term

## Description
This feature is added to prepare for the mega menu search feature UI

### CODE REVIEW

File: packages/api-client/src/next/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { searchProduct } from './product'
+
```
Comment (Fifciu) [inline packages/api-client/src/next/api/index.ts:2]: Redundant new line
```suggestion
```

File: packages/api-client/src/next/api/product/search.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            }
+        }
+    )
+
+    const response = await context.client.apolloClient.query({
+        query: gql(products.query),
+        variables: products.variables
+    })
+
+    return response?.data ?? []
```
Comment (Fifciu) [inline packages/api-client/src/next/api/product/search.ts:45]: Please make sure you are not cutting off there pagination/filters/sorting data

File: packages/composables/src/useSearch/params.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Context } from '../types'
+import { ProductsSearchParams, UseSearchFactoryParams } from '@vue-storefront/core'
+import { ProductConnection } from '@vue-storefront/shopify-api'
+
+export const useSearchFactoryParams: UseSearchFactoryParams<ProductConnection, ProductsSearchParams> = {
+    search (context: Context, params: ProductsSearchParams): Promise<ProductConnection> {
```
Comment (Fifciu) [inline packages/composables/src/useSearch/params.ts:6]: Spaces are inconsistent. In one place you use 2, in other 4. Let's make it consistent (we use 2 spaces).

File: packages/theme/pages/Search.vue [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    const { search, result } = useSearch('products');
+    const products = computed(() => result.value);
+
+    watchEffect(() => {
+      console.log(result.value);
+    });
+
+    onSSR(async () => {
+      await search({
+        term: 'b',
```
Comment (Fifciu) [inline packages/theme/pages/Search.vue:29]: Should this component be a part of the PR? Or maybe it should be updated?

File: packages/theme/pages/Search.vue [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import { useSearch } from '@vue-storefront/shopify';
+
+export default defineComponent({
+  setup() {
+    const { search, result } = useSearch('products');
+    const products = computed(() => result.value);
+
+    watchEffect(() => {
+      console.log(result.value);
+    });
```
Comment (Fifciu) [inline packages/theme/pages/Search.vue:25]: Looks like something supposed to be removed
```suggestion
```

**Comment by aureate-labs-team** [comment]
Please resolve conflicts and open conversations

**Pull request merged**